doc/sphinx/proof-engine/ltac.rst

```diff
@@ -407,221 +408,242 @@ We can branch with the following structure:
 
    Branching is left-associative.
 
-First tactic to work
-~~~~~~~~~~~~~~~~~~~~
+First tactic to succeed
+~~~~~~~~~~~~~~~~~~~~~~~
 
-Backtracking branching may be too expensive. In this case we may
-restrict to a local, left biased, branching and consider the first
-tactic to work (i.e. which does not fail) among a panel of tactics:
+Backtracking branching may be too expensive. We can restrict
+backtracking to local, left biased branching and use the first
+tactic that succeeds from a list of tactics:
 
-.. tacn:: first [{*| @ltac_expr}]
+.. tacn:: first [ {*| @ltac_expr } ]
    :name: first
 
    The :n:`@ltac_expr__i` are evaluated to :n:`v__i` and :n:`v__i` must be
    tactic values for i = 1, ..., n. Supposing n > 1,
-   :n:`first [@ltac_expr__1 | ... | @ltac_expr__n]` applies :n:`v__1` in each
+   :tacn:`first` :n:`[@ltac_expr__1 | ... | @ltac_expr__n]` applies :n:`v__1` in each
    focused goal independently and stops if it succeeds; otherwise it
    tries to apply :n:`v__2` and so on. It fails when there is no
    applicable tactic. In other words,
-   :n:`first [@ltac_expr__1 | ... | @ltac_expr__n]` behaves, in each goal, as the first
+   :tacn:`first` :n:`[@ltac_expr__1 | ... | @ltac_expr__n]` behaves, in each goal, as the first
    :n:`v__i` to have *at least* one success.
 
+   :tacn:`first` is an :token:`ltac_expr1`.
+
    .. exn:: No applicable tactic.
       :undocumented:
 
+   .. todo the following is not accepted as a regular tactic but it does seem to do something
+      see https://github.com/coq/coq/pull/12103#discussion_r422249862.
+      Probably the same thing as for the :tacv:`solve` below.
+      The code is in Coretactics.initial_tacticals
+
    .. tacv:: first @ltac_expr
 
       This is an |Ltac| alias that gives a primitive access to the first
       tactical as an |Ltac| definition without going through a parsing rule. It
-      expects to be given a list of tactics through a ``Tactic Notation``,
-      allowing to write notations of the following form:
+      expects to be given a list of tactics through a :cmd:`Tactic Notation` command,
+      permitting notations with the following form to be written:
 
       .. example::
 
          .. coqtop:: in
 
             Tactic Notation "foo" tactic_list(tacs) := first tacs.
```

Comment:
This example makes me realize that we should really find a better name for the `tactic_list` we introduced. The best alternative name I could find is: `goal_tactics`.